gridt/src/app/home/expandable/expandable.component.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { IonicModule } from '@ionic/angular';
+
+import { ExpandableComponent } from './expandable.component';
+import { By } from '@angular/platform-browser';
+
+describe('ExpandableComponent', () => {
+  let component: ExpandableComponent;
+  let fixture: ComponentFixture<ExpandableComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ ExpandableComponent ],
+      imports: [IonicModule.forRoot()]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ExpandableComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should toggle itemExpanded', () => {
```

Comment:
Good test.

Reply:
If you are testing this, it would be sensible to test that it also does _not_ do this when the text is too short.

Reply:
When i add this test:

```
  it('should toggleFullMessage on click', (done) => {
    spyOn(component, 'toggleFullMessage');
    component.signalMessage = "this message is long enough to require more space.";
    component.width = 5;

    let button = fixture.debugElement.query(By.css('.expand-wrapper')).nativeElement;
    button.click();

    fixture.whenStable().then(() => {
      expect(component.toggleFullMessage).toHaveBeenCalled();
      expect(component.itemExpanded).toBe(true, 'true after toggle');
    });
    done();
  });
```

it fails on this:

`expect(component.itemExpanded).toBe(true, 'true after toggle');`

But i don't know why

Reply:
The new toggle function:
```
  ngOnInit() {
    this.width = document.getElementById('signalText').offsetWidth;
  }

  toggleFullMessage() {
    if (this.signalMessage.length > this.width) {
      this.itemExpanded = !this.itemExpanded;
    }
  }
```

Reply:
https://hatoum.com/blog/2016/11/12/jasmine-unit-testing-dont-forget-to-callthrough